Community/Tdarr_Plugin_MC93_Migz1FFMPEG.js

```diff
@@ -335,8 +364,72 @@ const plugin = (file, librarySettings, inputs, otherArguments) => {
     response.preset = '-c:v vp8_cuvid';
   }
 
-  response.preset += `${genpts}, -map 0 -c:v hevc_nvenc -cq:v 19 ${bitrateSettings} `
-  + `-spatial_aq:v 1 -rc-lookahead:v 32 -c:a copy -c:s copy -max_muxing_queue_size 9999 ${extraArguments}`;
+  // Determine if GPU is present and which gpu to use based on utilization of each GPU
+  response.infoLog += 'Running nvidia-smi to see if a GPU is available and selecting the GPU with less utilization\n';
+  let gpu_num = -1;
+  let gpu_util = 100000;
+  let result_util = 0;
+  let gpu_count = '';
+	let gpus_to_exclude = inputs.exclude_gpus.toLowerCase().split(",");
+  try {
+    let gpu_res = execSync('nvidia-smi --query-gpu=name --format=csv,noheader');
+    gpu_res = gpu_res.toString().trim();
+    gpu_res = gpu_res.split(/\r?\n/);
+    /* When nvidia-smi returns an error it contains 'nvidia-smi' in the error
+      Example: Linux: nvidia-smi: command not found
+               Windows: 'nvidia-smi' is not recognized as an internal or external command, 
+                   operable program or batch file. */
+    if (!gpu_res[0].includes('nvidia-smi')) {
+      gpu_count = gpu_res.length;
+    } else {
+      gpu_count = -1;
+    }
+  } catch (error) {
+    response.infoLog += 'Error in reading nvidia-smi output!\n';
+    response.infoLog += error.message;
+    gpu_count = -1;
+  }
+  if (gpu_count > 0) {
+    for (let gpui = 0; gpui < gpu_count; gpui++) {
+      // Check if GPU name is in GPUs to exclude
+      const is_gpu_excluded = gpus_to_exclude.filter(gpu_name => {
+        try {
+          if (gpu_count[i].toLowerCase().includes(gpu_name.toLowerCase())) {
+            return true;
+          }
+        } catch (error) {}
+        return false;
+      });
+      if (is_gpu_excluded.length >= 1) {
+        response.infoLog += `GPU ${gpui} is in exclusion list, will not be used!\n`;
+      } else {
+        result_util = parseInt(execSync(
+          `nvidia-smi --query-gpu=utilization.gpu --format=csv,noheader,nounits -i ${gpui}`), 10);
+        if (!Number.isNaN(result_util)) { // != "No devices were found") {
+          response.infoLog += `GPU ${gpui} : Utilization ${result_util}%\n`;
+          if (result_util < gpu_util) {
+            gpu_num = gpui;
+            gpu_util = result_util;
+          }
+        } else {
+          result_util = execSync(`nvidia-smi --query-gpu=utilization.gpu --format=csv,noheader,nounits -i ${gpui}`);
```

Comment:
Could be a single execSync to retrieve it and only parseInt on line 408.

Reply:
Updated it to wrap it in `try catch` like this:

> try {
          const cmd_gpu = `nvidia-smi --query-gpu=utilization.gpu --format=csv,noheader,nounits -i ${gpui}`;
          result_util = parseInt(execSync(cmd_gpu), 10);
          if (!Number.isNaN(result_util)) { // != "No devices were found") {
            response.infoLog += `GPU ${gpui} : Utilization ${result_util}%\n`;
            if (result_util < gpu_util) {
              gpu_num = gpui;
              gpu_util = result_util;
            }
          }
        } catch (error) {
          response.infoLog += `Error in reading GPU ${gpui} Utilization\nError: ${error}\n`;
        }

Reply:
`try catch` will also be able to catch any errors while executing `execSync`